lib/Parse/ParseDecl.cpp

```diff
@@ -2406,10 +2406,24 @@ Decl *Parser::ParseDeclarationAfterDeclaratorAndAttributes(
           StopTokens.push_back(tok::r_paren);
         SkipUntil(StopTokens, StopAtSemi | StopBeforeMatch);
         Actions.ActOnInitializerError(ThisDecl);
-      } else
+      } else {
+        if (D.getDeclSpec().isUnpackSpecified()) {
+          // TODO: make more robust, add asserts, etc.
+          auto TpeVars = D.getDeclSpec().typeVariables();
+          assert(TpeVars.size() == 1 && "Expected exactly one type variable in _Unpack specifier");
```

Comment:
Please issue a diagnostic instead of asserting.  This will crash the compiler if the user has more than 1 type variable.

Reply:
Done. Added a test as well.